From 50cd68b994d988c116731a1ff36f3537c6103890 Mon Sep 17 00:00:00 2001
From: Tony Asleson <tasleson@redhat.com>
Date: Wed, 12 Feb 2020 10:04:58 -0600
Subject: [PATCH 1/1] Specify signed char

x86 architecture default to signed characters.  Other architectures
default to unsigned characters.  When building on non-x86 arch we
get the following error

./json.hpp:22700:42:   required from here
./json.hpp:8494:24: error: comparison is always true due to limited
                           range of data type [-Werror=type-limits]
 8494 |             if ('\x00' <= c and c <= '\x1F')
      |                 ~~~~~~~^~~~

Signed-off-by: Tony Asleson <tasleson@redhat.com>
---
 c_binding/json.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/c_binding/json.hpp b/c_binding/json.hpp
index 6b6655a..a4ed769 100644
--- a/c_binding/json.hpp
+++ b/c_binding/json.hpp
@@ -3097,7 +3097,7 @@ scan_number_done:
     std::size_t chars_read = 0;
 
     /// raw input token string (for error messages)
-    std::vector<char> token_string {};
+    std::vector<signed char> token_string {};
 
     /// buffer for variable-length tokens (numbers, strings)
     string_t token_buffer {};
-- 
2.32.0

